src/irmin-http/swagger.yml

```diff
@@ -0,0 +1,126 @@
+swagger:  "2.0"
+info:
+  title: "Irmin REST API"
+  description: "This is the first draft of a REST API for the
+    [irmin](https://github.com/mirage/irmin/) database"
+  version: "0.0.1"
+  contact:
+    name: "David Udelson"
+    email: "dru5@cornell.edu"
+paths:
+  /branches:
+    get:
+      summary: "list all branches"
+      consumes:
+      - "application/json"
+      produces:
+      - "application/json"
+      responses:
+        200:
+          description: "<response description here>" # TODO
+        404:
+          description: "" #TODO
+    post:
+      summary: "create a new branch"
+      responses:
+        200:
+          description: "" # TODO
+        201:
+          description: "" # TODO
+  /branch/{branchName}:
+    get:
+      summary: "get a branch by name"
+      responses:
+        200:
+          description: "" #TODO
+        404:
+          description: "" #TODO
+    delete:
+      summary: "delete a branch by its ID"
+      responses:
+        200:
+          description: "" #TODO
+        404:
+          description: "" #TODO
+  /commits:
+    get:
+      summary: "lists all commits"
+      responses:
+        200:
+          description: "" #TODO
+        404:
+          description: "" #TODO
+  /commits/{commitHash}:
+    get:
+      summary: "get a commit by its hash"
+      responses:
+        200:
+          description: "" #TODO
+        404:
+          description: "" #TODO
+  /paths:
```

Comment:
I think the summary was poorly written, sorry about that. I was intending `/path` to be the way to get the value associated with a key. I believe `key`s are specified by `path`s internally, so I used this name to be consistent with the internal naming, although I guess it could just as easily be `/key`.

The current API has low-level endpoints for blobs, trees, commits, and branches. If I understand correctly, the blobs and trees are internal details that probably have no place being exposed in a high-level API. The `/path` endpoint is intended to replace this functionality by replicating what the command-line currently does. For example, from the command-line you can do this:

    $ irmin write my/var foo
    $ irmin read my/var
    foo

I was thinking the equivalent for the REST API could be this:

    $ curl -d '{"key": "my/var", "value": "foo"}' http://localhost:8080/v1/paths
    # 201 created
    $ curl http://localhost:8080/v1/path/my/var
    # 200 OK. value = foo

I'm not sure exactly what the json return schema look like yet, but that's the general idea.